src/TestData/Factories/DonationFactory.php

```diff
@@ -97,6 +102,40 @@ public function getDonationCurrency() {
 		return $this->currency;
 	}
 
+
+	/**
+	 * @param string $date
+	 */
+	public function setDonationStartDate( $date ) {
+		$this->startDate = $date;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getDonationDate() {
+		if ( ! $this->isValidDate( $this->startDate ) ) {
+			return $this->faker->dateTimeThisYear()->format( 'Y-m-d H:i:s' );
+		}
+
+		return $this->faker->dateTimeBetween( $this->startDate, $endDate = 'now' )->format( 'Y-m-d H:i:s' );
+	}
+
+	/**
+	 * @param string $date
+	 *
+	 * @return bool
+	 */
+	public function isValidDate( $date ) {
```

Comment:
I think this could be simplified using the `DateTime` class. Not a blocker, but it would help to leverage these sort of objects to provide structure to the data and dates are an easy place to start.

Reply:
Resolved cde0350